web/src/app/monitor/services/restConnector/rest-connector.service.ts

```diff
@@ -1,28 +1,106 @@
 import { Injectable } from '@angular/core';
-import { HttpClient, HttpHeaders } from '@angular/common/http';
-import { environment } from '../../../../environments/environment';
+import { HttpClient } from '@angular/common/http';
+import { environment } from 'environments/environment';
 import { Observable } from 'rxjs';
 import { State } from 'app/models/state';
-import { WorkbasketCounter } from 'app/monitor/models/workbasket-counter';
+import { ReportData } from '../../models/report-data';
+import { ChartData } from 'app/monitor/models/chart-data';
+import { TaskanaDate } from 'app/shared/util/taskana.date';
 
 @Injectable()
 export class RestConnectorService {
 
   constructor(private httpClient: HttpClient) {
   }
 
-  getTaskStatistics(): Observable<State[]> {
-    return this.httpClient.get<Array<State>>(environment.taskanaRestUrl + '/v1/monitor/countByState?states=READY,CLAIMED,COMPLETED')
+  getTaskStatusReport(): Observable<ReportData> {
+    return this.httpClient.get<ReportData>(environment.taskanaRestUrl + '/v1/monitor/tasks-status-report?states=READY,CLAIMED,COMPLETED')
   }
 
+  getWorkbasketStatistics(): Observable<ReportData> {
+    return this.httpClient.get<ReportData>(environment.taskanaRestUrl
+      + '/v1/monitor/tasks-workbasket-report?daysInPast=5&states=READY,CLAIMED,COMPLETED');
+  }
+
+  getClassificationTasksReport(): Observable<ReportData> {
+    return this.httpClient.get<ReportData>(environment.taskanaRestUrl
+      + '/v1/monitor/tasks-classification-report');
+  }
+
+  getChartData(source: ReportData): Array<ChartData> {
+    const result = new Array<ChartData>();
+
+    Object.keys(source.rows).forEach(key => {
+      const rowData = new ChartData();
+
+      rowData.label = key;
+      rowData.data = new Array<number>();
+
+      source.meta.header.forEach((headerValue: string) => {
+        rowData.data.push(source.rows[key].cells[headerValue]);
+      })
 
-  getWorkbasketStatistics(): Observable<WorkbasketCounter> {
-    return this.httpClient.get<WorkbasketCounter>(environment.taskanaRestUrl
-      + '/v1/monitor/taskcountByWorkbasketDaysAndState?daysInPast=5&states=READY,CLAIMED,COMPLETED')
+      result.push(rowData)
+    })
+
+    return result;
+  }
+
+  getChartHeaders(source: ReportData, dateFormat = false): Array<string> {
+    const result = new Array<string>();
+    source.meta.header.forEach((header: string) => {
+      result.push(this.parseClassificationHeadersText(header, dateFormat));
+    })
+    return result;
   }
 
-  getTaskStatusReport(): Observable<Object> {
-    return this.httpClient.get(environment.taskanaRestUrl + '/v1/monitor/taskStatusReport')
+  parseClassificationHeadersText(header: string, dateFormat = false): string {
+    if (dateFormat) {
+      header = this.dateParseHeaders(header);
+    } else {
+      header = this.defaultParseHeaders(header);
+    }
+    return header;
   }
 
+  defaultParseHeaders(header: string): string {
```

Comment:
Since we agreed that the ReportData from the rest service provides the display name, i don't like that it is modified. I'd rather prefer a new class which extends `TimeIntervalColumnHeader` where the `displayName` would be overwritten.

Reply:
Ok, I didn't see displayName from TimeIntervarlColumnHeader, I will extend it and remove the transformation in client side.